### PR TITLE
rafaelmassimo_start_to_work_on_the_favorite_city

### PR DESCRIPTION
**Pull request by rafaelmassimo**
So far this code is for save the city from the searc bar and save as favorite

### CODE REVIEW

File: js/search_bar.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    favorite_cities = JSON.parse(localStorage.getItem("city"))
+    favorite_cities.map(city => {
+        const newOption = document.createElement("option");
+        newOption.innerText = city.trim()
+        newOption.value = city.trim().toLowerCase()
+        chooseFavoriteCity.append(newOption)
+    })
+}
+
+console.log(citiesLocalStorage)
```
Comment (qbixmex) [inline js/search_bar.js:17]: Remove this console log

**Review by qbixmex — COMMENTED** [review]
Check my comment

**Review by qbixmex — DISMISSED** [review]
Well done, everything is fine

File: js/search_bar.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+let chooseFavoriteCity = document.getElementById("favorite-cities")
+let favorite_cities = [];
+const citiesLocalStorage = localStorage.getItem("city")
+
+if (citiesLocalStorage) {
+    favorite_cities = JSON.parse(localStorage.getItem("city"))
+    favorite_cities.map(city => {
+        const newOption = document.createElement("option");
```
Comment (small-valley) [inline js/search_bar.js:8]: How about make new function which receive one string argument?
Because from line 21 you do the same thing.

**Review by small-valley — CHANGES_REQUESTED** [review]

File: js/search_bar.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        star_clicked.style.fill = "";
+    }
+});
+
+
+
+
+
+
+// let new_city_event = document.getElementById("search-city");
```
Comment (small-valley) [inline js/search_bar.js:62]: If you don't need these codes, please delete.

**Review by small-valley — CHANGES_REQUESTED** [review]

File: js/search_bar.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    newOption.value = city.trim().toLowerCase(); //Here on the string will be applied the trim and the lower case because the value has to be in this way
+    chooseFavoriteCity.append(newOption)
+}
+if (citiesLocalStorage) {
+    favorite_cities = JSON.parse(localStorage.getItem("city"));
+    favorite_cities.forEach(city => {
+        saveAsFavorite(city);
+    });
+}
+
```
Comment (qbixmex) [inline js/search_bar.js:24]: Remove Extra Line

File: js/search_bar.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    const newCityValue = new_city_event.value.trim();
+
+    if (newCityValue !== "" && !favorite_cities.includes(newCityValue)) {
+        favorite_cities.push(newCityValue);
+        localStorage.setItem("city", JSON.stringify(favorite_cities));
+        saveAsFavorite(newCityValue);
+        new_city_event.value = "";
+    }
+})
+
```
Comment (qbixmex) [inline js/search_bar.js:40]: Remove Exta Lines

File: js/search_bar.js (hunk 3 above)
Comment (qbixmex) [reply js/search_bar.js]: Yes remove unnecessary lines and commented code

**Review by small-valley — APPROVED** [review]
Great Job!!

**Review by qbixmex — APPROVED** [review]

**Pull request merged**